File: java/samples/batch-mode/src/main/java/com/myscript/iink/samples/batchmode/JsonResult.java
/*
 * Copyright (c) MyScript. All rights reserved.
 */

package com.myscript.iink.samples.batchmode;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class JsonResult {
    @SerializedName("events")
    private ArrayList<Stroke> strokes;

    JsonResult(ArrayList<Stroke> strokes) {
        this.strokes = strokes;
    }

    ArrayList<Stroke> getStrokes() {
        return strokes;
    }

    @Override
    public String toString() {
        StringBuilder result = new StringBuilder("{\"events\":[");

        for (int i = 0; i < this.strokes.size(); i++) {
            result.append(this.strokes.get(i).toString()).append(i == this.strokes.size() - 1 ? "" : ",");
        }

        result.append("]}");

        return result.toString();
    }
}

File: java/samples/batch-mode-kt/src/main/java/com/myscript/iink/samples/batchmode/JsonResult.kt
package com.myscript.iink.samples.batchmode

import com.google.gson.annotations.SerializedName


class JsonResult {

    @SerializedName("events")
    private var strokes: ArrayList<Stroke>? = null

    fun JsonResult(strokes: ArrayList<Stroke>?) {
        this.strokes = strokes
    }

    fun getStrokes(): ArrayList<Stroke>? {
        return strokes
    }

    override fun toString(): String {
        val result = StringBuilder("{\"events\":[")
        for (i in 0 until strokes!!.size) {
            result.append(strokes!![i].toString())
                .append(if (i == strokes!!.size - 1) "" else ",")
        }
        result.append("]}")
        return result.toString()
    }
}
